src/idl_gen_csharp.cpp

```diff
@@ -37,6 +37,11 @@ static CommentConfig comment_config = {
 };
 
 namespace csharp {
+struct FieldInfo {
```

Comment:
These names are way to generic.. call this `FieldArrayLength` or something. Stick it inside `CsharpGenerator`